### ISSUE TITLE
Resize a video whilst keeping the controls over the video

### ISSUE THREAD

**Issue by rickpbaker** [drewbaker/wp-easy#29]
When on a 'reels' page and using a MP4 file instead of video, you can't use object-fit:contain without the video player controls appearing 'off' the video. Need to use an 'intrinsic scale' type solution:

https://www.npmjs.com/package/intrinsic-scale

**Comment by rsm0128** [comment]
@rickpbaker Could you provide me an example, please?

**Comment by rickpbaker** [comment]
You can see how this here:

https://resinprojects.com/reel/

![Screen Shot 2024-10-08 at 12 21 12](https://github.com/user-attachments/assets/fb65f901-31f5-402d-b4bf-bfc18694c375)

**Comment by rsm0128** [comment]
Please try to use `height: auto` instead of `height: 100%`

```
.template-reel .reel-video {
    height: auto;
    width: 100%;
    object-fit: contain;
}
```

**Comment by rickpbaker** [comment]
That makes the player controls behave correctly, but it means the video doesn't scale when you reduce the browser height.

![Screenshot 2024-10-08 at 9 40 42 PM](https://github.com/user-attachments/assets/631cd5f3-0097-4067-946d-f605af412d5d)

![Screenshot 2024-10-08 at 9 41 08 PM](https://github.com/user-attachments/assets/7070d5e0-47a3-43af-b9e5-8eb865f9bb72)

**Comment by rsm0128** [comment]
`max-height: 100%` will make the video scale when browser height is reduced. Though there will be spaces in left and right.
![image](https://github.com/user-attachments/assets/c413281a-d928-4e91-a634-b496d0a27d89)

**Comment by rickpbaker** [comment]
Yeah it's not the space that's the issue. The issue is the player controls being wider than the video. Doesn't look nice.

**Comment by drewbaker** [comment]
I have an idea about how we could do this using something like this: https://www.npmjs.com/package/intrinsic-scale and we put a data attribute on an element that will resize it to fit it's parent.

We should think about the syntax for this, and how other things like it. For example "when in view" (intersection observes) etc... 

```
<div>
    <video data-fit-to-parent/>
</div>
```

Maybe a better name for it...

**Comment by rsm0128** [comment]
@rickpbaker Please check and try below CSS code.
```
.template-reel .video-wrapper {
    height: 100%;
    max-width: 1200px;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    align-items: center;
}
.template-reel .reel-video {
    max-height: 100%;
    max-width: 100%;
    object-fit: contain;
    height: auto;
    width: auto;
}
```

**Comment by drewbaker** [comment]
I'll take this one, I know what needs to be done.
